OSPSuite.Presentation.Importer/Views/IColumnMappingControl.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using OSPSuite.Assets;
+using OSPSuite.Presentation.Importer.Core.DataFormat;
+using OSPSuite.Presentation.Importer.Presenters;
+using OSPSuite.Presentation.Views;
+using OSPSuite.Utility.Reflection;
+
+namespace OSPSuite.Presentation.Importer.Views
+{
+   public class ColumnMappingViewModel : Notifier
+   {
+      public string MappingName { get; private set; }
+      private string _description;
+
+      public string Description 
+      {
+         get => _description;
+         set => SetProperty(ref _description, value);
+      }
+      private DataFormatParameter _source;
+      public DataFormatParameter Source 
+      {
+         get => _source;
+         set
+         {
+            _source = value;
+            Description = ColumnMappingFormatter.Stringify(value);
+         }
+      }
+      public int Icon { get; set; }
+      public enum ColumnType
+      {
+         Mapping,
+         MetaData,
+         GroupBy,
+         AddGroupBy
+      }
+      public ColumnType CurrentColumnType { get; set; }
+      public ColumnMappingViewModel(ColumnType columnType, string columnName, string description, DataFormatParameter source, int icon)
+      {
+         CurrentColumnType = columnType;
+         MappingName = columnName;
+         Description = description;
+         Source = source;
+         Icon = icon;
+      }
+   }
+
+   public static class ColumnMappingFormatter
+   {
+      private static readonly string _ignored = $"{ColumnMappingOption.DescriptionType.Ignored}";
+      public static string Ignored(IgnoredDataFormatParameter model = null)
+      {
+         if (model == null)
+            return _ignored;
+         return $"{_ignored},{model.ColumnName}";
+      }
+
+      public static string GroupBy(GroupByDataFormatParameter model)
+      {
+         return $"{ColumnMappingOption.DescriptionType.GroupBy},{model.ColumnName}";
+      }
+
+      public static string AddGroupBy(AddGroupByFormatParameter model)
+      {
+         return $"{ColumnMappingOption.DescriptionType.AddGroupBy},{model.ColumnName}";
+      }
+
+      public static string Mapping(MappingDataFormatParameter model)
+      {
+         return $"{ColumnMappingOption.DescriptionType.Mapping},{model.ColumnName},{model.MappedColumn.Name},{model.MappedColumn.SelectedUnit}";
+      }
+
+      public static string MetaData(MetaDataFormatParameter model)
+      {
+         return $"{ColumnMappingOption.DescriptionType.MetaData},{model.ColumnName},{model.MetaDataId}";
+      }
+
+      public static string MappingName(DataFormatParameter model)
+      {
+         switch (model)
+         {
+            case IgnoredDataFormatParameter _:
+               return Ignored();
```

Comment:
what's this ? Never used it before (Ignored())

Reply:
It is actually used in some places. The ColumnMappingFormatter is responsible for stringifying and parsing DataFormatParameter objects. The Ignore() is the concrete case for generating the description for an ignored DataFormatParameter.